src/serde_value_ext.rs

```diff
@@ -0,0 +1,154 @@
+use anyhow::Result;
+use async_graphql::Value as GraphQLValue;
+use indexmap::IndexMap;
+use serde_json::Value;
+
+use crate::mustache::Mustache;
+use crate::path::PathString;
+
+pub trait ValueExt {
+    fn render_value(&self, ctx: &impl PathString) -> Result<GraphQLValue>;
+}
+
+fn eval_types(value: &Value) -> Result<GraphQLValue> {
+    match value {
+        Value::Array(arr) => {
+            let mut out = Vec::new();
+            for v in arr {
+                let value = eval_types(v)?;
+                out.push(value);
+            }
+            Ok(async_graphql::Value::List(out))
+        }
+        Value::String(s) => {
+            let out = serde_json::from_str::<Value>(s.as_str());
```

Comment:
Doesn't `string_to_value` already returns the Value with all the nested parsing, why we need to do this again here?

And shouldn't it just be `serde_json::from_str::<GraphQLValue>(&s)`  instead of `string_to_value` and `eval_types`?

Reply:
If I do that every type becomes string ["{{foo.bar}}"] Will output string Instead of Number

Reply:
I've created a https://github.com/tailcallhq/tailcall/pull/1231/files to show what I mean. I removed `eval_types` and tests are still passing